### PR TITLE
Add prometheus metrics to track federation delays

### PR DESCRIPTION
**Pull request by richvdh**
Add a pair of federation metrics to track the delays in sending PDUs to/from particular servers.

### CODE REVIEW

File: synapse/config/federation.py [hunk 1]
```diff
@@ -85,4 +96,16 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
           - '::1/128'
           - 'fe80::/64'
           - 'fc00::/7'
+
+        # Report prometheus metrics on the age of PDUs being sent to and received from
+        # the following domains. This can be used to give an idea of "delay" on inbound
+        # and outbound federation, though be aware that any delay can be due to problems
+        # at either end or with the intermediate network.
```
Comment (clokep) [inline synapse/config/federation.py:103]: Should we include that the default is an empty list?

File: synapse/config/homeserver.py [hunk 2]
```diff
@@ -92,5 +92,4 @@ class HomeServerConfig(RootConfig):
         TracerConfig,
         WorkerConfig,
         RedisConfig,
-        FederationConfig,
```
Comment (clokep) [inline synapse/config/homeserver.py:95]: I was confused by this, but turns out it is duplicated above (on line 61).

**Review by clokep — APPROVED** [review]
Seems reasonable to me.

File: synapse/config/homeserver.py (hunk 2 above)
Comment (richvdh) [reply synapse/config/homeserver.py:95]: sorry yes, should have clarified that!

**Pull request merged**